xdsl/irdl/declarative_assembly_format.py

```diff
@@ -223,11 +230,12 @@ def parse(self, parser: Parser, state: ParsingState) -> None:
         state.attributes = res
 
     def print(self, printer: Printer, state: PrintingState, op: IRDLOperation) -> None:
-        if not op.attributes:
+        # assert False, (op.attributes | op.properties, op)
+        if not op.attributes and not op.properties:
             return
         if self.with_keyword:
             printer.print(" attributes")
-        printer.print_op_attributes(op.attributes)
+        printer.print_op_attributes(op.attributes | op.properties)
```

Comment:
What happens in MLIR if there are both properties and attributes with the same name?
I feel that we should raise an error, whatever MLIR does.

Reply:
MLIR silently overrides the extra attributes by the expected property in that case.
I'm okay with making it an error on our side